src/api/courses.ts

```diff
@@ -73,27 +73,78 @@ router.get(
   },
 );
 
+router.get("/:id/students", requireAuthentication, async (req, res) => {
+  try {
+  const courseId = req.params.id;
+  const course = await Course.findById(courseId);
+  
+  if (!course) {
+    return res.status(404).json({ error: "Course not found" });
+  }
+  
+  const authUser = req.user;
+  
+  if (
+    authUser.role !== "admin" &&
+    (authUser.role !== "instructor" || authUser._id !== course.instructorId)
```

Comment:
`authUser` shouldn't have a field called `_id`. Verify that this is correct.
This check should be part of the `requireAuthentication` filter.